components/api/src/test/java/com/hotels/styx/api/messages/FullHttpResponseTest.java

```diff
@@ -0,0 +1,405 @@
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hotels.styx.api.messages;
+
+import com.google.common.collect.Iterables;
+import com.hotels.styx.api.HttpCookie;
+import com.hotels.styx.api.HttpResponse;
+import io.netty.buffer.ByteBuf;
+import io.netty.handler.codec.http.HttpResponseStatus;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+import rx.observers.TestSubscriber;
+
+import java.util.Optional;
+
+import static com.google.common.net.MediaType.ANY_AUDIO_TYPE;
+import static com.hotels.styx.api.HttpCookie.cookie;
+import static com.hotels.styx.api.HttpCookieAttribute.domain;
+import static com.hotels.styx.api.HttpCookieAttribute.maxAge;
+import static com.hotels.styx.api.HttpCookieAttribute.path;
+import static com.hotels.styx.api.HttpHeader.header;
+import static com.hotels.styx.api.HttpHeaderNames.CONTENT_LENGTH;
+import static com.hotels.styx.api.HttpHeaderNames.LOCATION;
+import static com.hotels.styx.api.matchers.HttpHeadersMatcher.isNotCacheable;
+import static com.hotels.styx.support.matchers.IsOptional.isAbsent;
+import static com.hotels.styx.support.matchers.IsOptional.isValue;
+import static io.netty.buffer.Unpooled.copiedBuffer;
+import static io.netty.handler.codec.http.HttpResponseStatus.BAD_GATEWAY;
+import static io.netty.handler.codec.http.HttpResponseStatus.BAD_REQUEST;
+import static io.netty.handler.codec.http.HttpResponseStatus.CREATED;
+import static io.netty.handler.codec.http.HttpResponseStatus.GATEWAY_TIMEOUT;
+import static io.netty.handler.codec.http.HttpResponseStatus.MOVED_PERMANENTLY;
+import static io.netty.handler.codec.http.HttpResponseStatus.MULTIPLE_CHOICES;
+import static io.netty.handler.codec.http.HttpResponseStatus.NO_CONTENT;
+import static io.netty.handler.codec.http.HttpResponseStatus.OK;
+import static io.netty.handler.codec.http.HttpResponseStatus.SEE_OTHER;
+import static io.netty.handler.codec.http.HttpResponseStatus.TEMPORARY_REDIRECT;
+import static io.netty.handler.codec.http.HttpVersion.HTTP_1_0;
+import static io.netty.handler.codec.http.HttpVersion.HTTP_1_1;
+import static java.lang.String.valueOf;
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.emptyIterable;
+import static org.hamcrest.Matchers.hasItem;
+import static org.hamcrest.Matchers.is;
+import static org.hamcrest.Matchers.not;
+import static org.hamcrest.Matchers.nullValue;
+import static org.hamcrest.collection.IsIterableContainingInOrder.contains;
+
+public class FullHttpResponseTest {
+    @Test
```

Comment:
We need to have a test to ensure that all headers, status & other response attributes are converted correctly to streaming type:

eg.

```
assertThat(streaming.status(), is(response.status()));
assertThat(streaming.headers(), is(response.headers());
...
```